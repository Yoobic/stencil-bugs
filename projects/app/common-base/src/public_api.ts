export { AppCommonBaseModule } from './lib/app-common-base.module';

export * from './lib/bases/app-chat-page/app-chat-page.component';
export * from './lib/bases/app-entity-action-page/app-entity-action-page.component';
export * from './lib/bases/app-form-dynamic-page/app-form-dynamic-page.component';
export * from './lib/bases/app-grid-dialog-page/app-grid-dialog-page.component';
export * from './lib/bases/app-menu-page/app-menu-page.component';
export * from './lib/bases/app-root/app-root.component';
export * from './lib/bases/app-search-page/app-search-page.component';
export * from './lib/bases/app-walkthrough-page/app-walkthrough-page.component';
export * from './lib/bases/app-whatsnew-page/app-whatsnew-page.component';
export * from './lib/bases/base-page/base-page.component';

export * from './lib/components/app-profile/app-profile.component';
export * from './lib/components/app-login/app-login.component';
export * from './lib/components/app-grid/app-grid.component';
export * from './lib/components/app-form-dynamic/app-form-dynamic.component';

export * from './lib/services/dialog/dialog.service';
export * from './lib/services/utils/utils.service';

export * from './lib/animations/animations';
export * from './lib/animations/transitions';

export * from './lib/modals/app-result-dialog-page/app-result-dialog-page.component';