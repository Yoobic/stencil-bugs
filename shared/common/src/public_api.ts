export { CommonModule } from './lib/common.module';

export * from './lib/decorators/mixin/mixin.decorator';
export * from './lib/decorators/persistent/persistent.decorator';
export * from './lib/directives/long-press/long-press.directive';

export * from './lib/interfaces/animate-state/animate-state.class';
export * from './lib/interfaces/animations/animations.class';
export * from './lib/interfaces/colors/colors.interface';
export * from './lib/interfaces/config-constants/config-constants.interface';
export * from './lib/interfaces/position/position.interface';

export * from './lib/pipes/array-from-number/array-from-number.pipe';
export * from './lib/pipes/cloudinary/cloudinary.pipe';
export * from './lib/pipes/date-format/date-format.pipe';
export * from './lib/pipes/file-size/file-size.pipe';
export * from './lib/pipes/highlight/highlight.pipe';
export * from './lib/pipes/https/https.pipe';
export * from './lib/pipes/keys/keys.pipe';
export * from './lib/pipes/limit-to/limit-to.pipe';
export * from './lib/pipes/map-values/map-values.pipe';
export * from './lib/pipes/newline/newline.pipe';
export * from './lib/pipes/nohtml/nohtml.pipe';
export * from './lib/pipes/order-by/order-by.pipe';
export * from './lib/pipes/round/round.pipe';
export * from './lib/pipes/safe/safe.pipe';
export * from './lib/pipes/take/take.pipe';
export * from './lib/pipes/time-ago/time-ago.pipe';
export * from './lib/pipes/timer/timer.pipe';

export * from './lib/services/config-constants/config-constants.service';
export * from './lib/services/core-config/core-config.service';
export * from './lib/services/dom-adapter/dom-adapter.service';
export * from './lib/services/exif-restorer/exif-restorer.service';
export * from './lib/services/geo-location/geo-location.service';
export * from './lib/services/local-forage/local-forage.service';
export * from './lib/services/local-storage/local-storage.service';
export * from './lib/services/log/log.service';
export * from './lib/services/network/network.service';
export * from './lib/services/promise/promise.service';
export * from './lib/services/resize/resize.service';
export * from './lib/services/ruler/ruler.service';
export * from './lib/services/utils/utils.service';

export * from './lib/mixins/lodash-mixins/lodash-mixins';