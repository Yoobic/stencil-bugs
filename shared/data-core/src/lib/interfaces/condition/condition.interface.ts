import { IFormField, MOBILE_FORM_FIELDS_ALL, FormFieldType, ICondition, IConditionalField } from '@shared/interfaces';

import { Editable } from '../../decorators/editable/editable.decorator';
import { Model } from '../../decorators/model/model.decorator';
import { CONDITION_TYPES } from './icondition.interface';
import { ResponseObject } from '../../interfaces/response-object/response-object.interface';

import { map } from 'lodash-es';

export const ROLES = [
    'dashboard', 'admin', 'manager', 'team', 'teamplus', 'creator', 'service', 'supervisor', 'quora', 'kiosk', 'score', 'nochat', 'anonymous', 'stat', 'todo',
    'serviceuser', 'polluser', 'newsuser', 'newscreator', 'documentsuser', 'calendaruser', 'store', 'clientadmin', 'missionanalysis', 'missionviewer', 'followup', 'followupnouser', 'profilenoedit',
    'workplace', 'trial', 'videocall', 'academy', 'pharmaone', 'instagram'
];

export const ROLES_ASK = [
    'manager', 'creator', 'quora', 'academy', 'academyplus'
];

export const ROLES_CONDITIONS = {
    isAdmin: { type: 'roles', operator: 'in', group: ['admin'] },
    isClientAdmin: { type: 'roles', operator: 'in', group: ['clientadmin'] },
    isAdminOrClientAdmin: { type: 'roles', operator: 'in', group: ['admin', 'clientadmin'] },
    isNotAdmin: { type: 'roles', operator: 'nin', group: ['admin'] },
    isManager: { type: 'roles', operator: 'in', group: ['manager'] },
    isTeam: { type: 'roles', operator: 'in', group: ['team'] },
    isWorkplace: { type: 'roles', operator: 'in', group: ['admin', 'workplace'] },
    isTrial: { type: 'roles', operator: 'in', group: ['trial'] },
    isNotTrial: { type: 'roles', operator: 'nin', group: ['trial'] },
    hasTodo: { type: 'roles', operator: 'in', group: ['todo'] },
    hasService: { type: 'roles', operator: 'in', group: ['admin', 'service'] }
};

// export const CLIENTS_CONDITIONS = {
//     isNotGmi: { type: 'groups', operator: 'nin', group: ['gmi_france'] }
// };

let conditions = {
    isNotField: 'type!="field"',
    isField: 'type=="field"',
    isTag: 'type=="tags"',
    isGroup: 'type=="groups"',
    isSelect: 'field.type=="selectmulti" or field.type=="selectbuttonsmulti"',
    isToggle: 'field.type=="checkbox" or field.type=="toggle"',
    isAutocomplete: 'field.type=="select" or field.type=="selectbuttons" or field.type=="autocomplete"',
    isFieldSimple: 'type=="field" and (field.type=="text" or field.type=="email" or field.type=="number" or field.type=="formula" or field.type=="date" or field.type=="tel" or field.type=="time" or field.type=="range" or field.type=="starrating")',
    isFieldWithValues: 'type=="field" and (field.type=="checkbox" or field.type=="toggle" or field.type=="select" or field.type=="selectmulti" or field.type=="selectbuttons" or field.type=="selectbuttonsmulti" or field.type=="autocomplete" or field.type=="selectimage")'
};

export function getGroupsTransform(res: ResponseObject) {
    if (res.data && res.data.filter) {
        res.data = res.data.filter(g => ROLES.indexOf(g._id) < 0 && g.isRole !== true);
    }
    return res;
}

export function isNotInformationField(m) {
    return m.type !== FormFieldType.information;
}

export function getFormFieldValues() {
    return map(MOBILE_FORM_FIELDS_ALL.filter(isNotInformationField), 'type');
}

@Model({ className: 'Condition' })
export class Condition extends ICondition {
    @Editable('Condition', { type: FormFieldType.text, required: true })
    title: string;

    @Editable('Condition', { type: FormFieldType.autocomplete, translate: true, values: CONDITION_TYPES, required: true })
    type: string;

    @Editable('Condition', { type: FormFieldType.autocomplete, condition: conditions.isField, required: true, idAttributeName: 'name' })
    field?: IFormField;

    // { condition: conditions.isSelect, values: ['in', 'notin', 'containsall'] }
    // contains all : check that all values are present.
    // Deactivated for the moment to be compatible with the V2
    @Editable('Condition', {
        type: FormFieldType.autocomplete,
        required: true,
        multiple: false,
        translate: true,
        condition: 'type',
        conditionalValues: [
            { condition: conditions.isNotField, values: ['in', 'notin'] },
            { condition: conditions.isSelect, values: ['in', 'notin'] },
            { condition: conditions.isToggle, values: ['equals', 'notequals'] },
            { condition: conditions.isAutocomplete, values: ['equals', 'in', 'notin'] }
        ],
        defaultValues: ['equals', 'notequals', 'greaterthan', 'lessthan']
    })
    operator: string;

    @Editable('Condition', { type: FormFieldType.autocomplete, condition: conditions.isTag, tag: true, multiple: true, collectionName: 'locations', required: true })
    tags?: Array<string>;

    @Editable('Condition', { title: 'GROUPS', required: true, type: 'autocomplete', collectionName: 'groups', filters: [[{ field: 'isRole', operator: { _id: 'neq' }, value: true }, { field: 'type', operator: { _id: 'nin' }, value: ['plan'] }]], hiddenFields: ['isRole', 'type'], idOnly: true, mapTransform: getGroupsTransform, condition: conditions.isGroup, multiple: true, clearable: false })
    group?: Array<string>;

    @Editable('Condition', { type: FormFieldType.autocomplete, required: true, condition: conditions.isFieldWithValues, dynamicValues: 'field.values', defaultValues: ['true', 'false'], multiple: true })
    values: Array<any>;

    @Editable('Condition', { dynamicType: 'field.type', defaultType: FormFieldType.text, required: true, condition: conditions.isFieldSimple })
    value: any;
}

@Model({ className: 'ConditionalField' })
export class ConditionalField extends IConditionalField {

    @Editable('ConditionalField', { type: FormFieldType.text, readonly: true, title: 'FIELDIF' })
    fieldTitle: string;

    @Editable('ConditionalField', {
        type: FormFieldType.autocomplete,
        required: true,
        multiple: false,
        translate: true,
        title: 'FIELDIS',
        conditionalValues: [{ condition: conditions.isNotField, values: ['in', 'notin'] }, { condition: conditions.isSelect, values: ['in', 'notin'] }, { condition: conditions.isToggle, values: ['equals', 'notequals'] }, { condition: conditions.isAutocomplete, values: ['equals', 'in', 'notin'] }],
        defaultValues: ['equals', 'notequals', 'greaterthan', 'lessthan']
    })
    operator: string;

    @Editable('ConditionalField', { type: FormFieldType.autocomplete, required: true, condition: conditions.isFieldWithValues, dynamicValues: 'field.values', defaultValues: ['true', 'false'], multiple: true })
    values: Array<any>;

    @Editable('ConditionalField', { dynamicType: 'field.type', defaultType: FormFieldType.text, required: true, condition: conditions.isFieldSimple })
    value: any;

    @Editable('ConditionalField', { type: FormFieldType.autocomplete, required: true, title: 'FIELDTYPE', values: getFormFieldValues(), translate: true })
    newFieldType: string;

    @Editable('ConditionalField', { type: FormFieldType.text, required: true, title: 'TITLE' })
    newfieldTitle: string;

    @Editable('ConditionalField', { title: 'INSTRUCTIONS', type: FormFieldType.textarea })
    newfieldDescription: string;

    @Editable('ConditionalField', { title: 'MANDATORY', type: FormFieldType.checkbox })
    newFieldRequired: boolean;
}
