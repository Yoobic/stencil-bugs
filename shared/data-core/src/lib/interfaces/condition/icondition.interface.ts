export const CONDITION_TYPES = ['field', 'tags', 'groups'];
export const CONDITION_ALL_OPERATORS = ['equals', 'notequals', 'in', 'notin', 'greaterthan', 'lessthan'];
export const SIMPLE_FIELD_TYPES = ['text', 'email', 'number', 'date', 'tel', 'time', 'range', 'starrating'];
export const WITH_VALUES_FIELD_TYPES = ['checkbox', 'toggle', 'select', 'selectmulti', 'selectbuttons', 'selectbuttonsmulti', 'autocomplete'];

export interface ICondition {
    _id?: string;
    type?: string;
    operator?: string;
    title?: string;
    field?: any;
    tags?: Array<string>;
    group?: Array<string>;
    values?: Array<any>;
    value?: any;
    key?: string;
}
