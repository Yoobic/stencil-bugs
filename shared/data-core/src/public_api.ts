export { DataCoreModule } from './lib/data-core.module';

export * from './lib/decorators/editable/editable.decorator';
export * from './lib/decorators/mapping/mapping.decorator';
export * from './lib/decorators/model/model.decorator';
export * from './lib/decorators/searchable/searchable.decorator';

export * from './lib/interfaces/aggregate-log/aggregate-log.interface';
export * from './lib/interfaces/algorithm/algorithm.interface';
export * from './lib/interfaces/autorenew-config/autorenew-config.interface';
export * from './lib/interfaces/backup/backup.interface';
export * from './lib/interfaces/catalog/catalog.interface';
export * from './lib/interfaces/chart-definition/chart-definition.interface';
export * from './lib/interfaces/condition/condition.interface';
export * from './lib/interfaces/condition/icondition.interface';
export * from './lib/interfaces/constants/constants.interface';
export * from './lib/interfaces/currency/currency.interface';
export * from './lib/interfaces/dashboard/dashboard.interface';
export * from './lib/interfaces/entity/entity.interface';
export * from './lib/interfaces/field-selector/field-selector.interface';
export * from './lib/interfaces/file/file.interface';
export * from './lib/interfaces/folder/folder.interface';
export * from './lib/interfaces/group/group.interface';
export * from './lib/interfaces/kpi/kpi.interface';
export * from './lib/interfaces/location-type/location-type.interface';
export * from './lib/interfaces/location/location.interface';
export * from './lib/interfaces/mapping/mapping.interface';
export * from './lib/interfaces/mission-description/mission-description.interface';
export * from './lib/interfaces/model/model.interface';
export * from './lib/interfaces/notification-event/notification-event.interface';
export * from './lib/interfaces/notification/notification.interface';
export * from './lib/interfaces/operation/operation.interface';
export * from './lib/interfaces/product/product.interface';
export * from './lib/interfaces/response-object/response-object.interface';
export * from './lib/interfaces/slide/slide.interface';
export * from './lib/interfaces/sort/sort.interface';
export * from './lib/interfaces/tableau/tableau.interface';
export * from './lib/interfaces/tag-group/tag-group.interface';
export * from './lib/interfaces/tag/tag.interface';
export * from './lib/interfaces/tenant/tenant.interface';
export * from './lib/interfaces/todo/todo.interface';
export * from './lib/interfaces/translation/translation.interface';
export * from './lib/interfaces/user/user.interface';
export * from './lib/interfaces/workplace/workplace.interface';

export * from './lib/pipes/authorized/authorized.pipe';
export * from './lib/pipes/currency/currency.pipe';
export * from './lib/pipes/filter/filter.pipe';
export * from './lib/pipes/username/username.pipe';
export * from './lib/pipes/useremail/useremail.pipe';
export * from './lib/pipes/userinitial/userinitial.pipe';

export * from './lib/services/activity/activity.service';
export * from './lib/services/algorithms/algorithms.service';
export * from './lib/services/authentication/authentication.guard';
export * from './lib/services/authentication/authentication.service';
export * from './lib/services/authentication/current-session.resolver';
export * from './lib/services/backup/backup.service';
export * from './lib/services/box/box.service';
export * from './lib/services/blog/blog.service';
export * from './lib/services/links/links.service';
export * from './lib/services/broker/broker.service';
export * from './lib/services/config/config.service';
export { Dashboard as DashboardCore } from './lib/services/dashboard/dashboard.service';
export * from './lib/services/data-loader/data-loader.service';
export * from './lib/services/files/files.service';
export * from './lib/services/files/files-broker.service';
export * from './lib/services/googlemaps/googlemaps.service';
export * from './lib/services/googletranslate/googletranslate.service';
export * from './lib/services/http-interceptor/http-interceptor.service';
export * from './lib/services/loading-bar/loading-bar.service';
export * from './lib/services/locations/locations.service';
export * from './lib/services/loopback/loopback.service';
export * from './lib/services/models/models.service';
export * from './lib/services/print/print.service';
export * from './lib/services/push/push.service';
export * from './lib/services/requestor/requestor.service';
export * from './lib/services/security/security.service';
export * from './lib/services/session/session.service';
export * from './lib/services/smartloc/smartloc.service';
export * from './lib/services/users/users.service';
export * from './lib/services/version/version.service';
export * from './lib/services/workplace/workplace.service';
export * from './lib/services/xlsx/xlsx.service';

export * from './lib/validators/password-validator/password-validator';
export * from './lib/validators/username-validator/username-validator';
