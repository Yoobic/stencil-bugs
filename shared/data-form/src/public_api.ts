
export { DataFormModule } from './lib/data-form.module';

export * from './lib/interfaces/custom-form-field/custom-form-field.interface';
export * from './lib/interfaces/form-field/form-field.class';
export * from './lib/interfaces/gridstate/gridstate.interface';
export * from './lib/interfaces/cache/cache.interface';
export * from './lib/interfaces/custom-model/custom-model.interface';

export * from './lib/services/cell-renderer/cell-renderer.service';
export * from './lib/services/chart/chart.service';
export * from './lib/services/form-dynamic-builder/form-dynamic-builder.service';
export * from './lib/services/conditions/conditions.service';
export * from './lib/services/mapping/mapping.service';
export * from './lib/services/custom-models/custom-models.service';

export * from './lib/validators/form-filter-field-validator/form-filter-field-validator';
export * from './lib/validators/form-input-base-email-validator/form-input-base-email-validator';
export * from './lib/validators/form-input-base-number-validator/form-input-base-number-validator';
export * from './lib/validators/form-input-between-validator/form-input-between-validator';
export * from './lib/validators/form-input-checklist-validator/form-input-checklist-validator';
export * from './lib/validators/form-input-missionfield-validator/form-input-missionfield-validator';
export * from './lib/validators/form-input-multiphotos-validator/form-input-multiphotos-validator';
export * from './lib/validators/form-input-photo-validator/form-input-photo-validator';
export * from './lib/validators/form-input-select-validator/form-input-select-validator';
export * from './lib/validators/password-are-not-matching-validator/password-are-not-matching-validator';

export * from './lib/renderers/address/address-renderer.component';
export * from './lib/renderers/boolean/boolean-renderer.component';
export * from './lib/renderers/button/button-renderer.component';
export * from './lib/renderers/checklist/checklist-renderer.component';
export * from './lib/renderers/conformity-progress/conformity-progress-renderer.component';
export * from './lib/renderers/conformity-relative-progress/conformity-relative-progress-renderer.component';
export * from './lib/renderers/custom-model/custom-model-renderer.component';
export * from './lib/renderers/datetime/datetime-renderer.component';
export * from './lib/renderers/default/default-renderer.component';
export * from './lib/renderers/delta/delta-renderer.component';
export * from './lib/renderers/distance/distance-renderer.component';
export * from './lib/renderers/duration/duration-renderer.component';
export * from './lib/renderers/location/location-renderer.component';
export * from './lib/renderers/mission-status/mission-status-renderer.component';
export * from './lib/renderers/mission-validation/mission-validation-renderer.component';
export * from './lib/renderers/number/number-renderer.component';
export * from './lib/renderers/multiphotos/multiphotos-renderer.component';
export * from './lib/renderers/percentage/percentage-renderer.component';
export * from './lib/renderers/photo/photo-renderer.component';
export * from './lib/renderers/progress/progress-renderer.component';
export * from './lib/renderers/time/time-renderer.component';
export * from './lib/renderers/timer/timer-renderer.component';
export * from './lib/renderers/todo/todo-renderer.component';
export * from './lib/renderers/user-full-name/user-full-name-renderer.component';
export * from './lib/renderers/validation-progress/validation-progress-renderer.component';
export * from './lib/renderers/video/video-renderer.component';