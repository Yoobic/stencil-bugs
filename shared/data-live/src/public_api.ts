
export { DataLiveModule } from './lib/data-live.module';

export { Channel as ChannelInterface } from './lib/interfaces/channel/channel.interface';
export * from './lib/interfaces/channels/channels.interface';
export * from './lib/interfaces/message/message.interface';
export * from './lib/interfaces/message/message.model';

export * from './lib/services/channel/channel.service';
export * from './lib/services/intercom/intercom.service';
export * from './lib/services/messages/messages.service';
export * from './lib/services/pubnub/pubnub.service';
export * from './lib/services/track/track.service';
export * from './lib/services/twilio-token/twilio-token.service';
export * from './lib/services/raven-error-handler/raven-error-handler.service';