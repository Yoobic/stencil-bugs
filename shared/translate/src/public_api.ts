export { TranslateModule } from './lib/translate.module';

export * from './lib/pipes/distance/distance.pipe';
export * from './lib/pipes/filter/filter.pipe';
export * from './lib/pipes/polyglot/polyglot.pipe';

export * from './lib/services/translate/translate.service';