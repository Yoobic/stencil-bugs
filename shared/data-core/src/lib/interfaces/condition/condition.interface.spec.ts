import { Condition, ConditionalField } from './condition.interface';
import { Models } from '../../../public_api';

describe('shared: data-core: interfaces: Condition', () => {
    it('should create a condition entity', () => {
        let entity = new Condition();
        expect(entity._id).toBeUndefined();
        let model = Models.getModel('Condition');
        expect(model).toBeTruthy();
        expect(model).toMatchSnapshot();
    });
    it('should create a conditional field entity', () => {
        let entity = new ConditionalField();
        expect(entity._id).toBeUndefined();
        let model = Models.getModel('ConditionalField');
        expect(model).toBeTruthy();
        expect(model).toMatchSnapshot();
    });
});
