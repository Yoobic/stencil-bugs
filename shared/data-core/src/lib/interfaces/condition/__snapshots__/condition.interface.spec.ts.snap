// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared: data-core: interfaces: Condition should create a condition entity 1`] = `
IModel {
  "_formFields": Array [
    Object {
      "filterName": "title",
      "name": "title",
      "required": true,
      "title": "TITLE",
      "type": "text",
    },
    Object {
      "filterName": "type",
      "name": "type",
      "required": true,
      "title": "TYPE",
      "translate": true,
      "type": "autocomplete",
      "values": Array [
        "field",
        "tags",
        "groups",
      ],
    },
    Object {
      "condition": "type==\\"field\\"",
      "filterName": "field",
      "idAttributeName": "name",
      "name": "field",
      "required": true,
      "title": "FIELD",
      "type": "autocomplete",
    },
    Object {
      "condition": "type",
      "conditionalValues": Array [
        Object {
          "condition": "type!=\\"field\\"",
          "values": Array [
            "in",
            "notin",
          ],
        },
        Object {
          "condition": "field.type==\\"selectmulti\\" or field.type==\\"selectbuttonsmulti\\"",
          "values": Array [
            "in",
            "notin",
          ],
        },
        Object {
          "condition": "field.type==\\"checkbox\\" or field.type==\\"toggle\\"",
          "values": Array [
            "equals",
            "notequals",
          ],
        },
        Object {
          "condition": "field.type==\\"select\\" or field.type==\\"selectbuttons\\" or field.type==\\"autocomplete\\"",
          "values": Array [
            "equals",
            "in",
            "notin",
          ],
        },
      ],
      "defaultValues": Array [
        "equals",
        "notequals",
        "greaterthan",
        "lessthan",
      ],
      "filterName": "operator",
      "multiple": false,
      "name": "operator",
      "required": true,
      "title": "OPERATOR",
      "translate": true,
      "type": "autocomplete",
    },
    Object {
      "collectionName": "locations",
      "condition": "type==\\"tags\\"",
      "filterName": "tags",
      "multiple": true,
      "name": "tags",
      "required": true,
      "tag": true,
      "title": "TAGS",
      "type": "autocomplete",
    },
    Object {
      "clearable": false,
      "collectionName": "groups",
      "condition": "type==\\"groups\\"",
      "filterName": "group",
      "filters": Array [
        Array [
          Object {
            "field": "isRole",
            "operator": Object {
              "_id": "neq",
            },
            "value": true,
          },
          Object {
            "field": "type",
            "operator": Object {
              "_id": "nin",
            },
            "value": Array [
              "plan",
            ],
          },
        ],
      ],
      "hiddenFields": Array [
        "isRole",
        "type",
      ],
      "idOnly": true,
      "mapTransform": [Function],
      "multiple": true,
      "name": "group",
      "required": true,
      "title": "GROUPS",
      "type": "autocomplete",
    },
    Object {
      "condition": "type==\\"field\\" and (field.type==\\"checkbox\\" or field.type==\\"toggle\\" or field.type==\\"select\\" or field.type==\\"selectmulti\\" or field.type==\\"selectbuttons\\" or field.type==\\"selectbuttonsmulti\\" or field.type==\\"autocomplete\\" or field.type==\\"selectimage\\")",
      "defaultValues": Array [
        "true",
        "false",
      ],
      "dynamicValues": "field.values",
      "filterName": "values",
      "multiple": true,
      "name": "values",
      "required": true,
      "title": "VALUES",
      "type": "autocomplete",
    },
    Object {
      "condition": "type==\\"field\\" and (field.type==\\"text\\" or field.type==\\"email\\" or field.type==\\"number\\" or field.type==\\"formula\\" or field.type==\\"date\\" or field.type==\\"tel\\" or field.type==\\"time\\" or field.type==\\"range\\" or field.type==\\"starrating\\")",
      "defaultType": "text",
      "dynamicType": "field.type",
      "filterName": "value",
      "name": "value",
      "required": true,
      "title": "VALUE",
    },
  ],
  "appareance": Map {},
  "className": "Condition",
  "collectionName": undefined,
  "feathersService": undefined,
  "fields": undefined,
  "include": undefined,
  "isCustom": undefined,
  "mappingFields": Map {},
  "searchSubquery": undefined,
  "searchableFields": Array [],
  "type": [Function],
}
`;

exports[`shared: data-core: interfaces: Condition should create a conditional field entity 1`] = `
IModel {
  "_formFields": Array [
    Object {
      "filterName": "fieldTitle",
      "name": "fieldTitle",
      "readonly": true,
      "title": "FIELDIF",
      "type": "text",
    },
    Object {
      "conditionalValues": Array [
        Object {
          "condition": "type!=\\"field\\"",
          "values": Array [
            "in",
            "notin",
          ],
        },
        Object {
          "condition": "field.type==\\"selectmulti\\" or field.type==\\"selectbuttonsmulti\\"",
          "values": Array [
            "in",
            "notin",
          ],
        },
        Object {
          "condition": "field.type==\\"checkbox\\" or field.type==\\"toggle\\"",
          "values": Array [
            "equals",
            "notequals",
          ],
        },
        Object {
          "condition": "field.type==\\"select\\" or field.type==\\"selectbuttons\\" or field.type==\\"autocomplete\\"",
          "values": Array [
            "equals",
            "in",
            "notin",
          ],
        },
      ],
      "defaultValues": Array [
        "equals",
        "notequals",
        "greaterthan",
        "lessthan",
      ],
      "filterName": "operator",
      "multiple": false,
      "name": "operator",
      "required": true,
      "title": "FIELDIS",
      "translate": true,
      "type": "autocomplete",
    },
    Object {
      "condition": "type==\\"field\\" and (field.type==\\"checkbox\\" or field.type==\\"toggle\\" or field.type==\\"select\\" or field.type==\\"selectmulti\\" or field.type==\\"selectbuttons\\" or field.type==\\"selectbuttonsmulti\\" or field.type==\\"autocomplete\\" or field.type==\\"selectimage\\")",
      "defaultValues": Array [
        "true",
        "false",
      ],
      "dynamicValues": "field.values",
      "filterName": "values",
      "multiple": true,
      "name": "values",
      "required": true,
      "title": "VALUES",
      "type": "autocomplete",
    },
    Object {
      "condition": "type==\\"field\\" and (field.type==\\"text\\" or field.type==\\"email\\" or field.type==\\"number\\" or field.type==\\"formula\\" or field.type==\\"date\\" or field.type==\\"tel\\" or field.type==\\"time\\" or field.type==\\"range\\" or field.type==\\"starrating\\")",
      "defaultType": "text",
      "dynamicType": "field.type",
      "filterName": "value",
      "name": "value",
      "required": true,
      "title": "VALUE",
    },
    Object {
      "filterName": "newFieldType",
      "name": "newFieldType",
      "required": true,
      "title": "FIELDTYPE",
      "translate": true,
      "type": "autocomplete",
      "values": Array [
        "image",
        "photo",
        "multiphotos",
        "selectbuttons",
        "selectbuttonsmulti",
        "select",
        "selectmulti",
        "text",
        "textarea",
        "number",
        "starrating",
        "date",
        "time",
        "signature",
        "emailreport",
        "todo",
        "task",
        "checklist",
        "video",
        "audio",
        "document",
        "videoplayer",
        "barcode",
        "catalog",
        "selectimage",
        "autocomplete",
        "checkbox",
        "toggle",
        "tel",
        "email",
        "ranking",
        "range",
        "address",
        "location",
        "datetime-local",
        "timer",
        "formula",
      ],
    },
    Object {
      "filterName": "newfieldTitle",
      "name": "newfieldTitle",
      "required": true,
      "title": "TITLE",
      "type": "text",
    },
    Object {
      "filterName": "newfieldDescription",
      "name": "newfieldDescription",
      "title": "INSTRUCTIONS",
      "type": "textarea",
    },
    Object {
      "filterName": "newFieldRequired",
      "name": "newFieldRequired",
      "title": "MANDATORY",
      "type": "checkbox",
    },
  ],
  "appareance": Map {},
  "className": "ConditionalField",
  "collectionName": undefined,
  "feathersService": undefined,
  "fields": undefined,
  "include": undefined,
  "isCustom": undefined,
  "mappingFields": Map {},
  "searchSubquery": undefined,
  "searchableFields": Array [],
  "type": [Function],
}
`;
